napari/_vendor/qt_json_builder/qt_jsonschema_form/form.py

```diff
@@ -40,7 +40,8 @@ class WidgetBuilder:
             "enum": widgets.EnumSchemaWidget,
         },
         "integer": {
-            "spin": widgets.SpinSchemaWidget,
+            "spin": widgets.HighlightSizePreviewWidget,
```

Comment:
Could we instead try to have an approach where we can define the custom widget to use, within the NaparConfig class on the `_defaults.py` file. That way we do not need to change these code paths (yet).

Reply:
this was a bit of a cheat for now.  I was thinking we could talk about this today---about how to change this look up table for the different widgets.  Or how to approach.  I played a bit with it already, but haven't landed on anything.

Reply:
Sounds good. Let's talk about that today :)